aerogear/digger-installer
# proxy tasks

# OSX proxy setup

Adds OSX proxy config/setup if proxy_url is set. Needs sudo password to run.

## Changes

* created task provision-osx/configure_buildfarm_proxy.yml
* added task in provision-osx/main.yml

## How to verify

The following vars needs to be set:

```
proxy_host: $proxyhost
proxy_port: $proxyport
proxy_user: $proxy_user
proxy_pass: $proxy_pass
proxy_device: $proxy_device
proxy_ctx:
  - webproxy
  - securewebproxy
```

proxy_device var can be either "Wi-Fi" or "Ethernet" depends on mac machine setup. The following command be used to get a full list of devices: `networksetup -listallnetworkservices`

Both proxy_user and proxy_pass vars are optional.

```
 ansible-playbook -vvv -i my-cluster-up-example my-build-playbook.yml --tags osx_configure_proxy --ask-sudo-pass
```

This should set the mac osx global proxy, which can be verified in settings -> network -> device -> proxies.

The following commands should be used to verify the proxy config:

```
networksetup -getwebproxy
networksetup -getsecurewebproxy
```
